src/api/cvc4cpp.cpp

```diff
@@ -1204,5 +1209,159 @@ std::ostream& operator<<(std::ostream& out,
   return out;
 }
 
+/* -------------------------------------------------------------------------- */
+/* Solver                                                                     */
+/* -------------------------------------------------------------------------- */
+
+Solver::Solver(Options* opts)
+{
+  d_opts = std::unique_ptr<Options>(new Options());
```

Comment:
The order of data members in Solver is wrong: `d_opts` should be declared declared before `d_exprMgr` and before `d_smtEngine` otherwise we might run into troubles because `d_opts` gets destroyed before `d_smtEngine` and `d_exprMgr`. The initialization of `d_opts` can also be put in the initializer list.